src/components/create-safe/status/usePendingSafeCreation.ts

```diff
@@ -19,18 +24,42 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
   })
 }
 
-export const checkSafeCreationTx = async (provider: JsonRpcProvider, txHash: string) => {
+export const checkSafeCreationTx = async (provider: JsonRpcProvider, txHash: string, chainId: string) => {
   const TIMEOUT_TIME = 6.5
 
   try {
-    const receipt = await provider.waitForTransaction(txHash, 1, TIMEOUT_TIME * 60_000)
+    const blockNumber = await provider.getBlockNumber()
+    const txResponse = await provider.getTransaction(txHash)
+    const proxyContractAddress = getProxyFactoryContractInstance(chainId).address
+
+    const replacement = {
+      data: txResponse.data,
+      from: txResponse.from,
+      nonce: txResponse.nonce,
+      to: proxyContractAddress,
+      value: txResponse.value,
+      startBlock: txResponse.blockNumber || blockNumber,
+    }
+
+    const receipt = await provider._waitForTransaction(txHash, 1, TIMEOUT_TIME * 60_000, replacement)
 
     if (didRevert(receipt)) {
       return SafeCreationStatus.REVERTED
     }
 
     return SafeCreationStatus.SUCCESS
-  } catch (error) {
+  } catch (err) {
+    const error = err as EthersError
+
+    if (error.code === 'TRANSACTION_REPLACED') {
+      if (error.reason === 'cancelled') {
+        return SafeCreationStatus.ERROR
+      } else {
+        console.log('speed-up tx found!')
```

Comment:
If we intend to log this, can we provide more information, i.e. the transaction hash?

Reply:
Ah, good catch! It was a leftover. I will remove it